packages/block-editor/src/components/use-on-block-drop/index.js

```diff
@@ -259,10 +259,10 @@ export default function useOnBlockDrop( targetRootClientId, targetBlockIndex ) {
 		const files = getFilesFromDataTransfer( event.dataTransfer );
 		const html = event.dataTransfer.getData( 'text/html' );
 
-		if ( files.length ) {
-			_onFilesDrop( files );
-		} else if ( html ) {
+		if ( html ) {
```

Comment:
Could you just add a comment here about why the order is important and what this extra image in Windows is?

Reply:
@ellatrix Sure, I will add a comment. Do we need to change the order [here](https://github.com/WordPress/gutenberg/pull/30310/files#diff-5130f9f28d1f67e569b9c5498d443ab737c7e5be25e7219b7cf692c4dc3efa5dR26) as well?

Reply:
I guess that would be good if a manual test reveals a problem.

Reply:
@sathyapulse: I agree with Ella. Once you have fixed the order in the other handler, and added comments to each to justify the issue (possibly linking to this PR), let us know and we can merge this.

Reply:
Thank you, @mcsf and @ellatrix.

I've added the fix in the other handler and added comments [here](https://github.com/WordPress/gutenberg/pull/37151/commits/01dc086e26358b081d3ba808e6b5fb7616c7bd21). Could you please review it?